### PR TITLE
14 agrovoc

### LINKED ISSUE #14: Helper 'get_action' has not been defined
`get_action `helper method has been deprecated and returns a server error.

```
File '/usr/lib/ckan/default/src/ckanext-faociok/ckanext/faociok/templates/home/snippets/organizations.html', line 2 in top-level template code
  {% set groups = h.get_action('organization_list', {'limit': 4, 'all_fields': True, 'sort': 'package_count'}) or h.get_action('organization_list', {'limit': 4, 'all_fields': True}) %}
File '/usr/lib/ckan/default/lib/python2.7/site-packages/jinja2/environment.py', line 412 in getattr
  return obj[attribute]
File '/usr/lib/ckan/default/src/ckan/ckan/lib/helpers.py', line 72 in __getitem__
  key=key
HelperError: Helper 'get_action' has not been defined.
```

### PR DESCRIPTION
**Pull request by cezio**
fix https://github.com/geosolutions-it/C013-FAO-CIOK-CKAN/issues/14

### CODE REVIEW

**Comment by cezio** [comment]
> Too many logs during the ingestion of the vocabulary (a really infinite logging of the following), the meaning of the log should be to understand something clear on what is happening, not debug information everytime :)

done

> When adding multiple agrovoc terms and then saving the dataset, the error below occurs:

Yep, i missed information in Readme about `fao_agrovoc*` field in schema. You should add following line to schema.xml:

```
<dynamicField name="fao_agrovoc*" type="string" multiValued="true" indexed="true" stored="false"/>
```

Readme updated.

> Is not described a procedure to update an old vocabulary, it is important since the vocabulary is updated every month. The procedure and how it works in general meaning is useful.

There is no special/distinct metod to update vocabulary. You need to call `vocabulary import_agrovoc` with new data. That's it. I've added few lines in readme about this.

**Comment by cezio** [comment]
> Doing a DDI harvesting from Rwanda catalog it seems that the Region is no longer collected correctly:

Could you check that in different languages?

File: README.rst [hunk 1]
```diff
@@ -56,6 +58,33 @@ To install ckanext-faociok:
...
+#. Load AGROVOC vocabulary. Important note: `clean_agrovoc.sh` script is used to clean agrovoc file from unused triplets, so amount of data to parse is lower. Without it, ingestion script would use far more memory and time to process rdf file.::
+
+    cd files    
+    wget http://agrovoc.uniroma2.it/agrovocReleases/agrovoc_2018-09-03_lod.nt.zip
+    unzip agrovoc_2018-09-03_lod.nt.zip
+    mv agrovoc_2018-09-03_lod.nt agrovoc.nt
+    bash clean_agrovoc.sh agrovoc.nt
+    paster --plugin=ckanext-faociok vocabulary import_agrovoc agrovoc.clean.nt  --config=/etc/ckan/default/production.ini
+    
+.. note::
```
Comment (tdipisa) [inline README.rst]: Just one *Note* paragraph is enough

File: README.rst [hunk 2]
```diff
@@ -56,6 +58,33 @@ To install ckanext-faociok:
...
+
+    Mind that AGROVOC contains lot of data (over 30000 terms and around 500000 translated labels). File parsing and import will take ~10-20 minutes, depending on your hardware.
+
+.. note::
+
+    You should be able to update AGROVOC vocabulary just by providing to the script new version of RDF (mind that it should be in .nt format) from provided URLs.
+    Internally, each ingestion invocation removes existing terms and replaces with full set of new ones. This is processed within one transaction in database, so there should be no side-effects in running application (except for small slowdown for time of parsing and inserting new terms).
+
+.. note::
```
Comment (tdipisa) [inline README.rst]: If a Solr reindex is required please clearly list the steps to update the agrovoc vocabulary.

As agreed, after the update procedure of the agrovoc vocab, we should inform the admin in a log if some datasets have some agrovoc terms no longer part of the new ingested vocabulary:

`dataset {uuid} || {name} has some outdated agrovoc terms assigned: C_XXXXX, C_YYYYY `

Comment (cezio) [reply README.rst]: done

File: README.rst (hunk 1 above)
Comment (cezio) [reply README.rst]: done

**Comment by cezio** [comment]
>> Doing a DDI harvesting from Rwanda catalog it seems that the Region is no longer collected correctly:
> Could you check that in different languages?

This was actually because vocabulary query was looking in all terms without restricting to specific vocabulary, and we had `Rwanda` term in AGROVOC. This should be fixed now.

**Comment by cezio** [comment]
For the record, I've tested also earlier agrovoc files, but it seems they use different scheme for localized labels. Instead of skos#prefLabel, there's more complicated scheme using skos-xl#prefLabel to provide labels.

to illustrate both schemes, file from september:

```
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/2004/02/skos/core#Concept> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#inScheme> <http://aims.fao.org/aos/agrovoc> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#editorialNote> "Author: (Link 1807)"@en .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#broader> <http://aims.fao.org/aos/agrovoc/c_47299> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_ar_1299508902448> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_cs_1299508902465> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_en_1299508902483> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_es_1299508902505> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_fr_1299508902527> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_ja_1299508902551> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_pt_1299508902576> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_zh_1299508902604> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_tr_47326_1321795903159> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_de_699a6ade8089> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://purl.org/dc/terms/modified> "2011-11-20T21:31:43Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://purl.org/dc/terms/created> "2011-11-20T21:31:43Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://rdfs.org/ns/void#inDataset> <http://aims.fao.org/aos/agrovoc/void.ttl#Agrovoc> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@ar .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@cs .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@en .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@es .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@fr .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@ja .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@pt .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@zh .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais armigera"@tr .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#prefLabel> "Thais Armigera"@de .
<http://aims.fao.org/aos/agrovoc/c_47299> <http://www.w3.org/2004/02/skos/core#narrower> <http://aims.fao.org/aos/agrovoc/c_47326> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#broader> <http://aims.fao.org/aos/agrovoc/c_47299> .
```

file from july:
 

```
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/2004/02/skos/core#Concept> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#inScheme> <http://aims.fao.org/aos/agrovoc> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#editorialNote> "Author: (Link 1807)"@en .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2004/02/skos/core#broader> <http://aims.fao.org/aos/agrovoc/c_47299> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_ar_1299508902448> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_cs_1299508902465> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_en_1299508902483> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_es_1299508902505> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_fr_1299508902527> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_ja_1299508902551> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_pt_1299508902576> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_zh_1299508902604> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_tr_47326_1321795903159> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://www.w3.org/2008/05/skos-xl#prefLabel> <http://aims.fao.org/aos/agrovoc/xl_de_699a6ade8089> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://purl.org/dc/terms/modified> "2011-11-20T21:31:43Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
<http://aims.fao.org/aos/agrovoc/c_47326> <http://purl.org/dc/terms/created> "2011-11-20T21:31:43Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> .
```

Should we support both labeling schemes? It seems that skos#prefLabel is the new way.

**Pull request merged**